creds/lifetime.go

```diff
@@ -4,21 +4,32 @@ import (
 	"fmt"
 )
 
-// Constants for the session duration parameter in calls to sts:AssumeRole and
-// sts:GetSessionToken.
-const (
-	SessionLifetimeMin     = 900
-	SessionLifetimeMax     = 3600 * 12
-	SessionLifetimeDefault = 3600
-)
+// LifetimeLimits describes the minimum, maximum, and default values for
+// credential lifespan
+type LifetimeLimits struct {
+	Min, Max, Default int64
+}
+
+// SessionTokenLifetimeLimits describes the min, max, and default lifespan for
+// the sts:GetSessionToken call
+var SessionTokenLifetimeLimits = LifetimeLimits{Min: 900, Max: 3600 * 36, Default: 3600}
```

Comment:
It's perhaps a little weird that these limits are variables and not constants but that is pretty minor. Just Go issues.

Reply:
Yup :(

I wish these could be const, but this seemed like the cleanest way to do it